modules/packages/LinearAlgebra.chpl

```diff
@@ -0,0 +1,709 @@
+  var V: [Dom] eltType;
+  return V;
+}
+
+
+/* Return a vector (1D array) with domain and values of ``A`` */
+proc Vector(A: [?Dom] ?Atype, type eltType=Atype ) {
+  var V: [Dom] eltType = A;
+  return V;
+}
+
+
+pragma "no doc"
+proc Vector(x: ?t, Scalars...?n)  where isNumericType(t) {
+  type eltType = Scalars(1).type;
+  return Vector(x, (...Scalars), eltType=eltType);
+}
+
+
+/* Return a vector (1D array), given 2 or more numeric values
+
+   If `type` is omitted, it will be inferred from the first argument
+*/
+proc Vector(x: ?t, Scalars...?n, type eltType) where isNumericType(t) {
+
+  if !isNumeric(Scalars(1)) then
+      compilerError("Vector() expected numeric arguments");
+
+  var V: [0..n] eltType;
+
+  V[0] = x: eltType;
+
+  for i in 1..n do V[i] = Scalars[i]: eltType;
+
+  return V;
+}
+
+
+/* Return a matrix (2D array) over domain ``{0..#rows, 0..#cols}``*/
+proc Matrix(rows, cols, type eltType=real) where isIntegral(rows) && isIntegral(cols) {
+  var M: [0..#rows, 0..#cols] eltType;
+  return M;
+}
+
+
+/* Return a matrix (2D array) over domain ``Dom`` */
+proc Matrix(Dom, type eltType=real) where Dom.rank == 2 {
+  var M: [Dom] eltType;
+  return M;
+}
+
+
+/* Return a matrix (2D array) with domain and values of ``A`` */
+proc Matrix(A: [?Dom] ?eltType) where Dom.rank == 2 {
+  var M: [Dom] eltType = A;
+  return M;
+}
+
+
+pragma "no doc"
+proc Matrix(Arrays...?n) {
+  type eltType = Arrays(1).eltType;
+  return Matrix((...Arrays), eltType=eltType);
+}
+
+
+/*
+    Return a matrix (2D array), given 2 or more vectors. Vectors are
+    concatenated such that the ``ith`` vector corresponds to the matrix slice:
+    ``A[i, ..]``
+
+    If `type` is omitted, it will be inferred from the first argument
+*/
+proc Matrix(const Arrays...?n, type eltType) {
+  // TODO -- assert all array domains are same length
+  //         Can this be done via type query?
+
+  if Arrays(1).domain.rank != 1 then compilerError("Matrix() expected 1D arrays");
+
+  const dim2 = 0..#Arrays(1).domain.dim(1).length,
+        dim1 = 0..#n;
+
+  var M: [{dim1, dim2}] eltType;
+
+  for i in dim1 do
+    M[i, ..] = Arrays(i+1)[..]: eltType;
+
+  return M;
+}
+
+
+/* Return an identity matrix over domain ``{0..#m, 0..#n}`` */
+proc eye(m, n, type eltType=real) {
+  var A: [{0..#m, 0..#n}] eltType;
+  const diagonal = min(m, n);
+  for i in 0..#diagonal do A[i, i] = 1: eltType;
+  return A;
+}
+
+
+/* Return an identity matrix over domain ``Dom`` */
+proc eye(Dom: domain(2), type eltType=real) {
+  var A: [Dom] eltType;
+  const diagonal = min(Dom.shape(1), Dom.shape(2));
+  return eye(Dom.shape(1), Dom.shape(2), eltType=eltType);
+}
+
+
+//
+// Matrix Operations
+//
+
+pragma "no doc"
+/* Transpose vector (no-op) */
+inline proc transpose(A: [?Dom]) where Dom.rank == 1 {
+  return A;
+}
+
+
+pragma "no doc"
+/* Syntactic sugar for transpose(Vector) */
+proc _array.T where this.domain.rank == 1 { return transpose(this); }
```

Comment:
I've discussed this with @bradcray && @mppf . However, since dropping LAPACK support, transpose is the only Linear Algebra specific array method. If it's considered controversial, we could drop it for now (as it's just a syntactic sugar anyway), and have more thorough discussion before 1.16.

Anyway, I'll add this to the list of interesting topics in the PR description.